packages/components/src/components/domain/event/eventEnrolmentStatus/useEventEnrolmentStatus.tsx

```diff
@@ -0,0 +1,32 @@
+import { EnrolmentStatusLabel } from '../../../../constants/event-constants';
+import { useEventTranslation } from '../../../../hooks';
+import type { EventFields } from '../../../../types';
+
+export function getEnrolmentStatus(event: EventFields): EnrolmentStatusLabel {
+  const now = new Date();
+  const { remainingAttendeeCapacity, enrolmentStartTime, enrolmentEndTime } =
+    event;
+
+  // TODO: Add EnrolmentStatusLabel.queueable when we can resolve it
+  if (remainingAttendeeCapacity === 0) {
+    return EnrolmentStatusLabel.full;
+  } else if (enrolmentStartTime && new Date(enrolmentStartTime) > now) {
+    return EnrolmentStatusLabel.enrolmentNotStartedYet;
+  } else if (enrolmentEndTime && new Date(enrolmentEndTime) < now) {
+    return EnrolmentStatusLabel.enrolmentEnded;
+  } else if (enrolmentStartTime && enrolmentEndTime) {
```

Comment:
The source ticket's description says remainingAttendeeCapacity needs to be not null and larger than zero for enrolment to be considered available. To take this into account how about e.g. changing this to:
```typescript
} else if (enrolmentStartTime && enrolmentEndTime && remainingAttendeeCapacity > 0) {
```
?

Reply:
well, I can add that without the lt-statement, because otherwise I need to check the nullinesh also.

Reply:
I was thinking that as the type is `remainingAttendeeCapacity?: number | null` and as `(undefined > 0) === false` and `(null > 0) === false` and the only remaining possibilities through the type system are of type `number` using `remainingAttendeeCapacity > 0` here would handle also the undefined and null possibilities. But I think this is good enough already as it may be that the data in itself actually has an invariant of the value never being negative. So I leave it up to you whether you want to take that edge case into account or not.

Reply:
so, I did use `else if (enrolmentStartTime && enrolmentEndTime && remainingAttendeeCapacity)` , which checks the both, the nullness and zeroness.